search-sentiments-map/routes/search.js

```diff
@@ -26,22 +26,141 @@ const json = require('./../public/country-code.json');
+  const [customSearchTopic] = await datastore.runQuery(query);
+
+  let countriesDataToReturn = [];
+  let timestamp;
+  if (customSearchTopic.length !== 0 &&
+      Date.now() - customSearchTopic[0].timestamp <
+      CURRENT_SEARCH_RESULT_THRESHOLD_24_HOURS_MS) {
+        timestamp = customSearchTopic[0].timestamp;
+        let countriesToAddDataFor = [];
+        let countriesData = customSearchTopic[0].countries;
+
+        // Determine whether a country has existing data or data needs to be
+        // retrieved from Custom Search API for this country on this topic.
+        countries.forEach(country => {
+          let countryData = countriesData
+              .filter(countries => countries.country === country);
+          if (countryData.length === 0) {
+            countriesToAddDataFor.push(country);
+          } else {
+            countriesDataToReturn.push(countryData[0]);
+          }
+        });
+
+        // Obtain custom search data for countries without current data.     
+        // Add new custom search data to existing entity and to the data to
+        // send back to front end.
+        if (countriesToAddDataFor.length !== 0) {
+          let newCountriesData = await getSearchResultsForArrayOfCountries(
+            countriesToAddDataFor, topic);
+          await addNewCountryData(newCountriesData, customSearchTopic[0]);
+          countriesDataToReturn = countriesDataToReturn.concat(newCountriesData);
+        }
+  } else {
+    // Get data for all of the requested countries when there is no existing
+    // entity and create a new entity with this data.
+    countriesDataToReturn = await getSearchResultsForArrayOfCountries(
+      countries, topic);
+    await addCustomSearchTopicEntityToDatastore(topic, countriesDataToReturn);
+    timestamp = Date.now();
+  }
+
+  try {
+    let customSearchTopicJsonArray = {
+      topic: topic,
+      countries: countriesDataToReturn,
+      timestamp: timestamp,
+    };
+    return customSearchTopicJsonArray;
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/** 
+ * Retrieves search result data for given countries for a given topic. Returns
+ * this data.
+ * @param {Array} countries Array of 2 letter country codes for search results
+ *     to be written in.
+ * @param {string} topic Search query.
+ * @return {Object} Formatted object with country search result data and
+ *     country overall score for all given countries.
+ */
+async function getSearchResultsForArrayOfCountries(countries, topic) {
+  let countriesSearchResultData = [];
+  for (let i = 0; i < countries.length; i++) {
+    const countryResults =
+        await getCustomSearchResultsForCountry(countries[i], topic);
+    countriesSearchResultData.push({
+      country: countries[i],
+      averageSentiment: countryResults.score,
+      results: countryResults.results,
+    });
+  }
+  return countriesSearchResultData;
+}
+
+/** 
+ * Adds country data for specific topic to a CustomSearchTopic entity.
```

Comment:
{@code CustomSearchTopic}

Reply:
I just looked this up, apparently @code is deprecated. Will swap this and all previous/future comments on this to backticks.
https://google.github.io/styleguide/jsguide.html#appendices-documentation-annotations